core/src/main/java/com/linecorp/armeria/server/cors/CorsService.java

```diff
@@ -84,6 +86,27 @@ public static CorsServiceBuilder builder(Iterable<String> origins) {
         return new CorsServiceBuilder(copied);
     }
 
+    /**
+     * Returns a new {@link CorsServiceBuilder} with origins matching the {@code originPredicate}.
+     */
+    public static CorsServiceBuilder builder(Predicate<String> originPredicate) {
+        return new CorsServiceBuilder(originPredicate);
+    }
+
+    /**
+     * Returns a new {@link CorsServiceBuilder} with origins matching the {@code originRegex}.
+     */
+    public static CorsServiceBuilder builderForOriginRegex(String originRegex) {
+        return builderForOriginRegex(Pattern.compile(originRegex));
+    }
+
+    /**
+     * Returns a new {@link CorsServiceBuilder} with origins matching the {@code originRegex}.
+     */
+    public static CorsServiceBuilder builderForOriginRegex(Pattern originRegex) {
+        return new CorsServiceBuilder(originRegex);
+    }
```

Comment:
```suggestion
    public static CorsServiceBuilder builderForOriginRegex(Pattern originRegex) {
        requireNonNull(originRegex, "originRegex");
        return builder(origin -> originRegex.matcher(origin).matches());
    }
```
If we delegate here, we can remove the following constructors such as `CorsServiceBuilder(Pattern originRegex) {...}`.